spyder/plugins/editor/widgets/codeeditor/tests/test_multicursor.py

```diff
@@ -0,0 +1,174 @@
+    click_at(codeeditor, qtbot, 6, ctrl=True, alt=True)
+    # A cursor was added
+    assert bool(codeeditor.extra_cursors)
+    codeeditor.toggle_multi_cursor(False)
+    # Extra cursors removed on settings toggle
+    assert not bool(codeeditor.extra_cursors)
+    click_at(codeeditor, qtbot, 3, ctrl=True, alt=True)
+    # Extra cursors not added wnen settings "multi-cursor enabled" is False
+    assert not bool(codeeditor.extra_cursors)
+    click_at(codeeditor, qtbot, 13, ctrl=True, alt=True, shift=True)
+    # Column cursors not added wnen settings "multi-cursor enabled" is False
+    assert not bool(codeeditor.extra_cursors)
+
+
+def test_extra_selections_decoration(codeeditor, qtbot):
+    codeeditor.set_text("0123456789\n0123456789\n0123456789\n0123456789\n")
+    cursor = codeeditor.textCursor()
+    cursor.movePosition(QTextCursor.MoveOperation.Down,
+                        QTextCursor.MoveMode.MoveAnchor,
+                        3)  # column 0 row 4
+    codeeditor.setTextCursor(cursor)
+    click_at(codeeditor, qtbot, 6, ctrl=True, alt=True, shift=True)
+    selections = codeeditor.get_extra_selections("extra_cursor_selections")
+    assert len(selections) == 3
+
+
+def test_multi_cursor_verticalMovementX(codeeditor, qtbot):
+    # Ensure extra cursors (and primary cursor) keep column position when
+    #    moving up and down
+    codeeditor.set_text("012345678\n012345678\n\n012345678\n012345678\n")
+    click_at(codeeditor, qtbot, 4)
+    click_at(codeeditor, qtbot, 14, ctrl=True, alt=True)
+    for _ in range(3):
+        qtbot.keyClick(codeeditor, Qt.Key.Key_Down)
+    assert codeeditor.extra_cursors[0].position() == 25
+    assert codeeditor.textCursor().position() == 35
+    for _ in range(3):
+        qtbot.keyClick(codeeditor, Qt.Key.Key_Up)
+    assert codeeditor.extra_cursors[0].position() == 4
+    assert codeeditor.textCursor().position() == 14
+
+
+def test_overwrite_mode(codeeditor, qtbot):
+    # Multi-cursor rendering requires overwrite mode be handled manually as
+    #    there is no way to hide the primary textCursor with overwriteMode, and
+    #    there is no way to sync the blinking of extra cursors with native
+    #    rendering.
+    codeeditor.set_text("0123456789\n0123456789\n")
+    click_at(codeeditor, qtbot, 4)
+    qtbot.keyClick(codeeditor, Qt.Key.Key_Insert)
+    assert not codeeditor.overwriteMode()
+    assert codeeditor.overwrite_mode
+    qtbot.keyClick(codeeditor, Qt.Key.Key_A)
+    assert codeeditor.toPlainText() == "0123a56789\n0123456789\n"
+    click_at(codeeditor, qtbot, 16, ctrl=True, alt=True)
+    qtbot.keyClick(codeeditor, Qt.Key.Key_B)
+    assert codeeditor.toPlainText() == "0123ab6789\n01234b6789\n"
+    qtbot.keyClick(codeeditor, Qt.Key.Key_Insert)
+    assert not codeeditor.overwriteMode()
+    assert not codeeditor.overwrite_mode
+    qtbot.keyClick(codeeditor, Qt.Key.Key_C)
+    assert codeeditor.toPlainText() == "0123abc6789\n01234bc6789\n"
+
+# TODO test folded code
```

Comment:
The tests are still WIP.

Reply:
No problem. FYI, I'm ok with merging this PR as it is and wait for you to add these tests later. That would avoid the burden of constantly updating this PR when new changes are merged to `master`.

Reply:
@ccordoba12 Sounds good! I'm just getting back to my computer from holiday.. I worked a bit on the tests today which I'll submit, but I'll get this synced with master so it can be merged :)